node/network/statement-distribution/src/vstaging/tests/grid.rs

```diff
@@ -14,28 +14,2981 @@
+				&SigningContext { parent_hash: relay_parent, session_index: 1 },
+			)
+			.as_unchecked()
+			.clone();
+		let statement_d = state
+			.sign_statement(
+				v_d,
+				CompactStatement::Seconded(candidate_hash),
+				&SigningContext { parent_hash: relay_parent, session_index: 1 },
+			)
+			.as_unchecked()
+			.clone();
+
+		// Receive an advertisement from C.
+		{
+			let manifest = BackedCandidateManifest {
+				relay_parent,
+				candidate_hash,
+				group_index: other_group,
+				para_id: other_para,
+				parent_head_data_hash: pvd.parent_head.hash(),
+				statement_knowledge: StatementFilter {
+					seconded_in_group: bitvec::bitvec![u8, Lsb0; 0, 1, 1],
+					validated_in_group: bitvec::bitvec![u8, Lsb0; 0, 0, 0],
+				},
+			};
+
+			send_peer_message(
+				&mut overseer,
+				peer_c.clone(),
+				protocol_vstaging::StatementDistributionMessage::BackedCandidateManifest(
+					manifest.clone(),
+				),
+			)
+			.await;
+		}
+
+		// Should send a request to C.
+		{
+			let statements = vec![
+				statement_d.clone(),
+				state
+					.sign_statement(
+						v_e,
+						CompactStatement::Seconded(candidate_hash),
+						&SigningContext { parent_hash: relay_parent, session_index: 1 },
+					)
+					.as_unchecked()
+					.clone(),
+			];
+			assert_matches!(
+				overseer.recv().await,
+				AllMessages::NetworkBridgeTx(NetworkBridgeTxMessage::SendRequests(mut requests, IfDisconnected::ImmediateError)) => {
+					assert_eq!(requests.len(), 1);
+					assert_matches!(
+						requests.pop().unwrap(),
+						Requests::AttestedCandidateVStaging(mut outgoing) => {
+							assert_eq!(outgoing.peer, Recipient::Peer(peer_c.clone()));
+							assert_eq!(outgoing.payload.candidate_hash, candidate_hash);
+
+							let res = AttestedCandidateResponse {
+								candidate_receipt: candidate.clone(),
+								persisted_validation_data: pvd.clone(),
+								statements,
+							};
+							outgoing.pending_response.send(Ok(res.encode()));
+						}
+					);
+				}
+			);
+
+			assert_matches!(
+				overseer.recv().await,
+				AllMessages::NetworkBridgeTx(NetworkBridgeTxMessage::ReportPeer(p, r))
+					if p == peer_c && r == BENEFIT_VALID_STATEMENT
+			);
+			assert_matches!(
+				overseer.recv().await,
+				AllMessages::NetworkBridgeTx(NetworkBridgeTxMessage::ReportPeer(p, r))
+					if p == peer_c && r == BENEFIT_VALID_STATEMENT
+			);
+
+			assert_matches!(
+				overseer.recv().await,
+				AllMessages::NetworkBridgeTx(NetworkBridgeTxMessage::ReportPeer(p, r))
+					if p == peer_c && r == BENEFIT_VALID_RESPONSE
+			);
+		}
+
+		let hypothetical = HypotheticalCandidate::Complete {
+			candidate_hash,
+			receipt: Arc::new(candidate.clone()),
+			persisted_validation_data: pvd.clone(),
+		};
+		let membership = vec![(relay_parent, vec![0])];
+		answer_expected_hypothetical_depth_request(
+			&mut overseer,
+			vec![(hypothetical, membership)],
+			None,
+			false,
+		)
+		.await;
+
+		// Receive messages from Backing subsystem.
```

Comment:
just a comment typo but to clarify, these statements are being sent to the candidate backing subsystem.